app/background.cc

```diff
@@ -25,7 +25,7 @@ struct background_t {
 };
 
 // Internal video reader thread
-static void read_thread(std::shared_ptr<background_t> pbkd) {
+static void read_thread(background_t *pbkd) {
```

Comment:
This somewhat counter-acts the reference counting done by the `std::shared_ptr`, thus extra care should be taken, that there is always one live reference of the original `std::shared_ptr` while any code of the `read_thread` function executes; otherwise risking UB. If you really have to pass a plain pointer here, make that point to the `std::shared_ptr` and copy the actual `std::shared_ptr` inside the routine to ensure this required local copy to ensure correct object lifetimes.

Reply:
Err, that's not part of this PR? It's in #122.. happy to discuss over there :smile:

Reply:
Saw too late. But will do.